hostinfo/hostinfo.go

```diff
@@ -41,6 +44,9 @@ func GetEnvType() EnvType {
 }
 
 func getEnvType() EnvType {
+	if inTestCase() {
```

Comment:
probably worth a comment that this should be first in the list. Otherwise it'd be easy to see this failing when running Go binary tests under, say, knative, which is something I've done before.

Reply:
Done